app/src/main/java/com/github/sdpcoachme/location/autocomplete/LocationAutocompleteHandler.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.sdpcoachme.location.autocomplete
+
+import com.github.sdpcoachme.data.UserLocation
+import java.util.concurrent.CompletableFuture
+
+/**
+ * This interface is used to launch the activity or process that allows the user to select a location.
+ * The implementation should also handle the result of the activity or process and return a
+ * UserLocation object.
+ */
+interface LocationAutocompleteHandler {
+
+    /**
+     * Launch the activity or process that allows the user to select a location.
+     * Note this method can be called multiple times without issues, if for example the user cancels
+     * the operation and wants to try again. The method should return a different CompletableFuture
+     * instance each time it is called, and relaunches the process/activity necessary.
+     *
+     * @return A CompletableFuture that will be completed with the UserLocation selected by the user.
+     *        If the user cancels the operation, the CompletableFuture will fail with an
+     *        AutocompleteCancelledException. If there is an error, the CompletableFuture will fail
+     *        with an AutocompleteFailedException.
+     */
+    fun launch(): CompletableFuture<UserLocation>
+
+    // Used to handle places autocomplete activity errors
+    class AutocompleteFailedException(message: String? = null, cause: Throwable? = null) : Exception(message, cause) {
+        constructor(cause: Throwable) : this(null, cause)
+    }
+    class AutocompleteCancelledException(message: String? = null, cause: Throwable? = null) : Exception(message, cause) {
+        constructor(cause: Throwable) : this(null, cause)
+    }
```

Comment:
The constructors for those classes don't seem necessary and are never used, maybe not worth keeping them ?

Reply:
Those constructor are here to match exactly the interface of the Exception class (which is a Java class). See https://stackoverflow.com/questions/45162869/kotlin-throw-custom-exception for more info about why it is a good idea to keep them.